plugins/kafka/kafka.cpp

```diff
@@ -54,24 +58,80 @@ namespace KafkaPlugin
     // Static Methods (internal)
     //--------------------------------------------------------------------------
 
+#ifdef _CONTAINERIZED
+    static void applyConfigProps(IPropertyTree * props, const char* configName, RdKafka::Conf* configPtr)
+    {
+        if (props)
+        {
+            Owned<IPropertyTreeIterator> iter = props->getElements(configName);
+            std::string errStr;
+
+            ForEach(*iter)
+            {
+                IPropertyTree & entry = iter->query();
+                const char* name = entry.queryProp("@name");
+                const char* value = entry.queryProp("@value");
+
+                if (name && *name)
+                {
+                    if (stricmp(name, "metadata.broker.list") != 0)
+                    {
+                        if (value && *value)
+                        {
+                            if (configPtr->set(name, value, errStr) != RdKafka::Conf::CONF_OK)
+                            {
+                                DBGLOG("Kafka: Failed to set config param from entry %s: '%s' = '%s'; error: '%s'", configName, name, value, errStr.c_str());
+                            }
+                            else if (doTrace(traceKafka))
+                            {
+                                DBGLOG("Kafka: Set config param from entry %s: '%s' = '%s'", configName, name, value);
+                            }
+                        }
+                    }
+                    else
+                    {
+                        DBGLOG("Kafka: Setting '%s' ignored in config %s", name, configName);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Look for an optional configuration entry and apply any found parameters
+     * to a librdkafka configuration object.
+     *
+     * @param   configName          The name of the configuration key within plugins/kafka;
+     *                              it is not required for the key to exist
+     * @param   configPtr           A pointer to the configuration object that
+     *                              will receive any found parameters
+     */
+    static void applyConfig(const char* configName, RdKafka::Conf* configPtr)
+    {
+        if (configName && *configName && configPtr)
+        {
+            applyConfigProps(getGlobalConfigSP()->getPropTree("plugins/kafka"), configName, configPtr);
+            applyConfigProps(getComponentConfigSP()->getPropTree("plugins/kafka"), configName, configPtr);
```

Comment:
is there ever any expense with calling RdKafka::Conf->set ? And if so, would it be better to compile (merge) the global/component configs, and call applyConfigProps just once ?

Reply:
e.g. could do:
```
                Owned<IPropertyTree> globalCopy = createPTreeFromIPT(getGlobalConfigSP()->queryPropTree("plugins/kafka"));
                synchronizePTree(globalCopy, getComponentConfigSP()->queryPropTree("plugins/kafka"), false);
                applyConfigProps(globalCopy, configName, configPtr);
```

Reply:
There is very little expense in Conf->set; it is updating some C struct values. I would leave this version as-is, especially since it is not on a hot path (used only during setup, not when passing messages).